nym-vpn-core/crates/nym-vpn-lib-types/src/tunnel_state.rs

```diff
@@ -198,6 +205,117 @@ pub enum ErrorStateReason {
     Internal(String),
 }
 
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ClientErrorReason {
+    Firewall,
+    Routing,
+    SameEntryAndExitGateway,
+    InvalidEntryGatewayCountry,
+    InvalidExitGatewayCountry,
+    MaxDevicesReached,
+    BandwidthExceeded,
+    SubscriptionExpired,
+    Dns(String),
+    Api(String),
+    Internal(String),
+}
+
+impl From<ErrorStateReason> for ClientErrorReason {
+    fn from(value: ErrorStateReason) -> Self {
+        match value {
+            ErrorStateReason::SameEntryAndExitGateway => Self::SameEntryAndExitGateway,
+            ErrorStateReason::InvalidEntryGatewayCountry => Self::InvalidEntryGatewayCountry,
+            ErrorStateReason::InvalidExitGatewayCountry => Self::InvalidExitGatewayCountry,
+            ErrorStateReason::BadBandwidthIncrease => Self::Api(value.to_string()),
+            ErrorStateReason::SyncAccount(err) => Self::Api(err.to_string()),
+            ErrorStateReason::SyncDevice(err) => Self::Api(err.to_string()),
+            ErrorStateReason::RegisterDevice(err) => {
+                if err
+                    .message_id()
+                    .is_some_and(|id| id.contains(MAX_DEVICES_REACHED_MESSAGE_ID))
+                {
+                    Self::MaxDevicesReached
+                } else {
+                    Self::Api(err.to_string())
+                }
+            }
+            ErrorStateReason::RequestZkNym(err) => match err {
+                RequestZkNymErrorReason::VpnApi(e) => match e.message_id.as_ref() {
+                    Some(id) if id.contains(BANDWIDTH_LIMIT_REACHED_MESSAGE_ID) => {
+                        Self::BandwidthExceeded
+                    }
+                    Some(id) if id.contains(SUBSCRIPTION_EXPIRED_MESSAGE_ID) => {
+                        Self::SubscriptionExpired
+                    }
+                    _ => Self::Api(e.message),
```

Comment:
to keep enough info of the underlaying api error, we should relay both the `message` and `message_id` (if any), ie.
```
        if let Some(id) = &e.message_id {
            format!(f, "{}, ID [{}]", e.message, id)
        } else {
           e.message
        }
```